docs/troubleshooting.md

```diff
@@ -0,0 +1,333 @@
+* Is the module loaded?
+
+```sh
+node=compute-0
+
+oc get mc -l app.kubernetes.io/part-of=onload
+
+# Expect two healthy services
+oc debug -q node/$node -- chroot /host systemctl status '*sfc*'
+
+# Expect the onload-module image in local image store
+oc debug -q node/$node -- chroot /host podman image ls -a | grep onload
+
+# Expect the module is loaded without fatal messages
+oc debug -q node/$node -- lsmod | grep sfc
+oc debug -q node/$node -- dmesg -xT | grep -i sfc
+```
+
+## NFD selectors
+
+```sh
+# Expect PCI 1924 snippet to be present
+kubectl get NodeFeatureDiscovery -n openshift-nfd -o yaml
+
+# Expect all nodes with sfc cards
+kubectl get nodes -l feature.node.kubernetes.io/pci-1924.present
+```
+
+## Node host
+
+```sh
+# Set node name and OpenShift or Kubernetes (as implementations have incompatibilities)
+debug_node="oc debug node/compute-0 -q"
+#debug_node="kubectl debug node/compute-0 --image=busybox -q"
+
+# Expect storage space available
+$debug_node -- chroot /host df -h /opt
+
+# Expect matching Onload version number
+$debug_node ONLOAD_PRELOAD=/opt/onload/usr/lib64/libonload.so -- chroot /host /opt/onload/usr/bin/onload --version
+
+# Expect both `sfc` and `onload` kernel modules loaded
+$debug_node -- chroot /host lsmod | grep -e sfc -e onload
+
+# Expect clean logs for both modules after boot time
+$debug_node -- dmesg -xT | grep -ie sfc -e onload
+
+$debug_node -- chroot /host /opt/onload/usr/bin/onload_stackdump lots # or `filters` or `filter_table`
+$debug_node -- chroot /host /opt/onload/usr/bin/onload_mibdump llap
+```
+
+## Accelerated pods
+
+```sh
+pod=onload-sfnettest-client
+
+# Expect `amd.com/onload`
+kubectl get pod $pod -o jsonpath='{..resources.limits}'
+
+# Expect path to `libonload.so` unless `setPreload` in Onload CR is disabled
+kubectl exec $pod -- sh -c 'echo $LD_PRELOAD'
+
+# Expect `onload` binary and/or `libonload.so` libraries. Adjust if different mount paths set in Onload CR
+kubectl exec onload-sfnettest-client -- ls -R /opt/onload
+
+# Expect Onload version
+kubectl exec onload-sfnettest-client -- /opt/onload/usr/lib64/libonload.so
+
+# Expect a net1 interface for Multus networks
+kubectl get events --field-selector reason=AddedInterface
+
+# Expect `NetworkAttachmentDefinition` referencing accelerable hardware
+nad=ipvlan-bond0
+kubectl get net-attach-def $nad -o yaml
+```
+
+Further details on individuals applications are available from Onload tools on the [node host](#node-host).
+
+For more complex issues, debug versions of Onload may be built.
+See [Build: Onload Source & Onload User](../README.md#onload-source--onload-user).
+
+### Modules and devices already loaded or mounted
+
+Remove any old deployments, including:
+
+* Onload Operator deployments which were not cleanly removed, or
+* alternatives to KubernetesOnload such as direct filesystem install.
+
+A crude method:
+
+```sh
+node=
+kubectl debug node/$node -- chroot /host rmmod -v onload sfc_char sfc_resource
```

Comment:
`sfc_driverlink`? `sfc`? In the case of `sfc` it is worth mentioning that if the nic is used for any other traffic then they will lose connection until a driver is re-installed.